sdk/core/azure-core/src/http/logging_policy.cpp

```diff
@@ -30,22 +30,69 @@ std::string TruncateIfLengthy(std::string const& s)
   return s.substr(0, BeginLength) + Ellipsis + s.substr(length - EndLength, EndLength);
 }
 
-std::string GetRequestLogMessage(Request const& request)
+std::string GetRequestLogMessage(
+    Azure::Core::Http::LoggingPolicyOptions const& options,
+    Request const& request)
 {
   std::ostringstream log;
 
-  log << "HTTP Request : " << HttpMethodToString(request.GetMethod()) << " "
-      << request.GetUrl().GetAbsoluteUrl();
+  std::string url = request.GetUrl().GetAbsoluteUrl();
+  if (!url.empty())
+  {
+    auto qm = url.find('?');
+    if (qm != std::string::npos)
+    {
+      url = url.substr(0, qm);
+    }
+  }
+
+  {
+    auto separ = '?';
+    for (auto qparam : request.GetUrl().GetQueryParameters())
+    {
+      url += separ + qparam.first + '=';
+
+      if (options.AllowedHttpQueryParameters.find(qparam.first)
+          != options.AllowedHttpQueryParameters.end())
+      {
+        url += TruncateIfLengthy(qparam.second);
```

Comment:
Consider that query parameters will be encoded, it might be easier to digest for humans if we call Decode() on them, unless we want to show exactly what we sent as verbatin.

Reply:
I think that since we show in the URL form, we should not decode, but show what gets sent. However, AllowedHttpQueryParameters shouldn't be encoded, I'll fix that.

Reply:
On the other hand, that would require us to decode it every time we log, so for performance reasons it makes sense to store it as encoded. I'll update the naming.